### ISSUE TITLE
After CanDeactivate Guard, click of "back" button changes url but router does not navigate

### ISSUE THREAD

**Issue by corykon** [angular/angular#14822]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
After being blocked by a CanDeactivate guard, if the user clicks the browser back arrow: 
-the url will change
-the router does NOT attempt to navigate (so doesn't seem to run through the guard)

Maybe related to [12851](https://github.com/angular/angular/issues/12851#issuecomment-262520855)?

**Expected behavior**
On clicking the back button, we run through the guard code like normal

**Minimal reproduction of the problem with instructions**
[http://plnkr.co/edit/DZwn4C0F8HpAjr04a5Ji?p=preview](http://plnkr.co/edit/DZwn4C0F8HpAjr04a5Ji?p=preview)

**What is the motivation / use case for changing the behavior?**
We should always warn the user when they try to navigate away from unsaved changes, and certainly don't want to have the url become disconnected from what is on screen.

**Please tell us about your environment:**
Windows 10, VS2015 and VSCode, SystemJS

* **Angular version:** 2.0.X
2.4.8

* **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]
Only tried in Chrome
 
* **Language:** [all | TypeScript X.X | ES6/7 | ES5]
Typescript

* **Node (for AoT issues):** `node --version` =   
N/A

**Comment by DzmitryShylovich** [comment]
Duplicate of https://github.com/angular/angular/issues/13586

**Comment by DzmitryShylovich** [comment]
@pkozlowski-opensource duplicate
